src/Symfony/Component/VarDumper/Caster/SplCaster.php

```diff
@@ -27,30 +27,25 @@ class SplCaster
         \SplFileObject::READ_CSV => 'READ_CSV',
     );
 
-    public static function castArrayObject(\ArrayObject $c, array $a, Stub $stub, $isNested)
+    public static function castArrayObject($c, array $a, Stub $stub, $isNested)
```

Comment:
Now that this method supports more than just `ArrayObject`, it should be renamed, right?

Reply:
Or for BC, we can probably keep this method signature as is and create a `castArrayIterator` one and both calling a private one?